### PR TITLE
dcnm_vrf: Integration tests - WIP

### PR DESCRIPTION
**Pull request by nkshrishail**
##### SUMMARY
Adding integration tests for dcnm_vrf module, for create, attach and deploy operations.

##### ISSUE TYPE
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME
dcnm_vrf

##### ADDITIONAL INFORMATION
Most of the values are hardcoded as of now due to limitations. 
Will enhance the plays once we find solutions for some of the limitations.

### CODE REVIEW

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- name: Verify if fabric - vxlan-shrishail-green-field is deployed.
+  dcnm_rest:
+    method: GET
+    path: /rest/control/fabrics/vxlan-shrishail-green-field
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that:
+      - 'result.response|length > 0'
```
Comment (chrisvanheuveln) [inline test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: You probably don't need this assert if you remove `ignore_errors` above.

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+---
```
Comment (chrisvanheuveln) [inline test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: Any reason to not put these three tests in the same test file?

File: test/integration/targets/dcnm_vrf/tests/nxapi/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    vrf_name: ansible-vrf-1
+    serial_numbers_vlans:
+      - 947BSAMCAU1: 210
+    deployment: False
+  register: result
+
+- assert:
+    that:
+      - 'result.changed == true'
+      - '"SUCCESS" in result.response.values()'
```
Comment (chrisvanheuveln) [inline test/integration/targets/dcnm_vrf/tests/nxapi/deploy.yaml]: Looks redundant. Will we ever get  `SUCCESS` with changed == false? 
or changed == true and it's not `SUCCESS`?

File: test/integration/targets/dcnm_vrf/tests/nxapi/attach.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+---
+- debug: msg="START connection={{ ansible_connection }}/create.yaml"
+
+- name: Verify if fabric - vxlan-shrishail-green-field is deployed.
+  dcnm_rest:
+    method: GET
+    path: /rest/control/fabrics/vxlan-shrishail-green-field
```
Comment (chrisvanheuveln) [inline test/integration/targets/dcnm_vrf/tests/nxapi/attach.yaml]: Please add a `# TBD` comment to the end of this line. I just want something to jump out when we do our final reviews.

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml (hunk 2 above)
Comment (nkshrishail) [reply test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: Just to keep operations separate, got inspired by some existing tests where I saw different files for different operations. Is there a style guideline for int tests?

File: test/integration/targets/dcnm_vrf/tests/nxapi/attach.yaml (hunk 4 above)
Comment (nkshrishail) [reply test/integration/targets/dcnm_vrf/tests/nxapi/attach.yaml]: Oh yeah, thanks for reminding, I will do that for all of these files.

File: test/integration/targets/dcnm_vrf/tests/nxapi/deploy.yaml (hunk 3 above)
Comment (nkshrishail) [reply test/integration/targets/dcnm_vrf/tests/nxapi/deploy.yaml]: Yeah, right.

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml (hunk 2 above)
Comment (chrisvanheuveln) [reply test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: The separate tests style predates my time with the team. When I came on most tests were combined in a single `sanity` test. But really it's whatever makes sense, which might be separate test files, especially if a given test operation has a lot of complicated tests.

Note there's also an overall CI duration cost for each separate test so if they can be combined that's usually preferred. For `network` I'm planning to combine them all in one file.

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml (hunk 1 above)
Comment (nkshrishail) [reply test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: I can take out the ignore_errors, but, still need to keep the assert. The GET API does not return error (400 or 500) when it does not find the fabric.

Comment (chrisvanheuveln) [reply test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: 👍

File: test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml (hunk 2 above)
Comment (nkshrishail) [reply test/integration/targets/dcnm_vrf/tests/nxapi/create.yaml]: All right, will move the tests to one file, sanity.yaml

**Pull request merged**